File: affy/sdk/calvin_files/writers/test/DataGroupHeaderWriterTest.cpp
////////////////////////////////////////////////////////////////
//
// Copyright (C) 2005 Affymetrix, Inc.
//
// This library is free software; you can redistribute it and/or modify
// it under the terms of the GNU Lesser General Public License 
// (version 2.1) as published by the Free Software Foundation.
// 
// This library is distributed in the hope that it will be useful, but
// WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
// or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License
// for more details.
// 
// You should have received a copy of the GNU Lesser General Public License
// along with this library; if not, write to the Free Software Foundation, Inc.,
// 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA 
//
////////////////////////////////////////////////////////////////

#include "DataGroupHeaderWriterTest.h"
#include "DataGroupHeaderWriter.h"
#include "ColumnInfo.h"

using namespace std;
using namespace affymetrix_calvin_io;

CPPUNIT_TEST_SUITE_REGISTRATION( DataGroupHeaderWriterTest );

void DataGroupHeaderWriterTest::setUp()
{
    writer = new DataGroupHeaderWriter();
}

void DataGroupHeaderWriterTest::tearDown()
{
    delete writer;
}

void DataGroupHeaderWriterTest::testCreation()
{
	DataGroupHeaderWriter w;
	CPPUNIT_ASSERT(1);
}

void DataGroupHeaderWriterTest::WriteTest()
{
	std::ofstream os;
    std::string f = "data_dataGroup_header";
    os.open(f.c_str(), std::ios::out|std::ios::binary|std::ios::trunc);
	if (!os)
	{
		CPPUNIT_ASSERT(0);
	}
    else
    {
	    DataGroupHeader hdr(L"dataGroup");
	    writer->Write(os, hdr);
        CPPUNIT_ASSERT(1);
        os.close();
    }

    //TODO: use data dataGroup header reader to read header and verify data
}

File: affy/sdk/java/AffxFusion/test/affymetrix/calvin/writers/DataGroupHeaderWriterTest.java
package affymetrix.calvin.writers;

import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.IOException;

import junit.framework.TestCase;
import affymetrix.calvin.data.DataGroupHeader;
import affymetrix.calvin.exception.UnsignedOutOfLimitsException;
import affymetrix.calvin.parsers.DataGroupHeaderReader;

public class DataGroupHeaderWriterTest extends TestCase {

	protected void setUp() throws Exception {
		super.setUp();
	}

	protected void tearDown() throws Exception {
		super.tearDown();
	}

	public void testWrite() throws FileNotFoundException, IOException, UnsignedOutOfLimitsException {
		DataGroupHeaderWriter writer = new DataGroupHeaderWriter();
		String f = "DataGroupHeader";
		BufferedFileOutputStream os = new BufferedFileOutputStream(f);
		DataGroupHeader grpHdr = new DataGroupHeader("DataGroup");
		writer.write(os, grpHdr);
		grpHdr.clear();
		os.close();
		assertTrue(true);

		// read header and verify data
		DataGroupHeaderReader reader = new DataGroupHeaderReader();
		FileInputStream is = new FileInputStream(f);
		reader.read(is, grpHdr);
		assertEquals(grpHdr.getName(), "DataGroup");
	}

}
